Objects/setobject.c

```diff
@@ -1380,17 +1482,26 @@ set_intersection_update_multi_impl(PySetObject *so, PyObject *args)
     tmp = set_intersection_multi_impl(so, args);
     if (tmp == NULL)
         return NULL;
+    Py_BEGIN_CRITICAL_SECTION(so);
     set_swap_bodies(so, (PySetObject *)tmp);
+    Py_END_CRITICAL_SECTION();
     Py_DECREF(tmp);
     Py_RETURN_NONE;
 }
 
 static PyObject *
 set_and(PySetObject *so, PyObject *other)
 {
+    PyObject *rv;
```

Comment:
I'd consider moving this declaration down to just above L1500, for a smaller diff.

Reply:
Ditto for `set_sub` and `set_isub`.